### ISSUE TITLE
False positive for consistent-destructuring

### ISSUE THREAD

**Issue by crystalfp** [sindresorhus/eslint-plugin-unicorn#2268]
In the code below, the rule suggest to use destructuring but does not suggest a way to fix the problem. In my opinion this is a perfectly valid destructuring and a possible change makes the code unclear. The code is:
```js
const structure = {
    atoms: [{position: [1,2,3]}]
};
const {position} = structure.atoms[0];
```
The rule complains to the last line. One way to satisfy the rule is to change the last line into:
```js
const {atoms} = structure;
const {position} = atoms[0];
```
That is much less clear that the first code.

consistent-destructuring

The example is above in the first section.

```js
const code = 'that should be ok';

const structure = {
    atoms: [{position: [1,2,3]}]
};
const {position} = structure.atoms[0];
```

**Comment by fregante** [comment]
That's not a false positive, it works as intended.

However we agree that the rule is not always ideal so it was recently removed from the "recommended" config:

- https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2260

**Comment by fregante** [comment]
Duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1501
